Zend/zend_inheritance.c

```diff
@@ -313,63 +447,20 @@ static zend_bool zend_do_perform_implementation_check(const zend_function *fe, c
 
 	for (i = 0; i < num_args; i++) {
 		zend_arg_info *fe_arg_info = &fe->common.arg_info[i];
+		zend_arg_info *proto_arg_info = (i < proto->common.num_args)
+			? &proto->common.arg_info[i]
+			: &proto->common.arg_info[proto->common.num_args];
 
-		zend_arg_info *proto_arg_info;
-		if (i < proto->common.num_args) {
-			proto_arg_info = &proto->common.arg_info[i];
-		} else {
-			proto_arg_info = &proto->common.arg_info[proto->common.num_args];
-		}
-
-		if (!zend_do_perform_arg_type_hint_check(fe, fe_arg_info, proto, proto_arg_info)) {
-			switch (ZEND_TYPE_CODE(fe_arg_info->type)) {
-				case IS_ITERABLE:
-					if (!zend_iterable_compatibility_check(proto_arg_info)) {
-						return 0;
-					}
-					break;
-
-				default:
-					return 0;
-			}
-		}
-
-		// This introduces BC break described at https://bugs.php.net/bug.php?id=72119
-		if (ZEND_TYPE_IS_SET(proto_arg_info->type) && ZEND_TYPE_ALLOW_NULL(proto_arg_info->type) && !ZEND_TYPE_ALLOW_NULL(fe_arg_info->type)) {
-			/* incompatible nullability */
-			return 0;
-		}
-
-		/* by-ref constraints on arguments are invariant */
-		if (fe_arg_info->pass_by_reference != proto_arg_info->pass_by_reference) {
+		if (!_check_inherited_parameter_type(fe, fe_arg_info, proto, proto_arg_info)) {
```

Comment:
Shouldn't this be passing on the `-1` state as well?

Reply:
Or maybe not? I don't really get how or at which point this code interacts with `CG(unverified_types)`. I expected that something would remove a class from that set if verification returns 1 (rather than -1), but don't see it.